src/cogent3/app/composable.py

```diff
@@ -937,6 +955,46 @@ def _validate_data_type(self, data):
 __app_registry = {}
 
 
+def _class_from_func(func):
+    """make a class based on func
+
+    Notes
+    -----
+    produces a class consistent with the necessary properties for
+    the define_app class decorator.
+
+    func becomes a static method on the class
+    """
+    # these methods MUST be in function scope so that separate instances are
+    # created for each decorated function
+    def _init(self, *args, **kwargs):
+        self._args = args
+        self._kwargs = kwargs
+
+    def _main(self, arg, **kwargs):
+        kw_args = deepcopy(self._kwargs)
+        kw_args = {**kw_args, **kwargs}
+        args = (arg,) + deepcopy(self._args)
```

Comment:
If we have more than one argument, we need `(*args,)`

Reply:
I am not sure about this, mainly because at present I have not made a decision about what the call api should support. This needs to be revisited.